src/rules/no-space-in-emphasis.js

```diff
@@ -34,32 +29,8 @@ const whitespacePattern = /[ \t]/u;
  */
 function createMarkerPattern(checkStrikethrough) {
```

Comment:
Can we also remove the function as part of this refactoring as no regular expression is dynamically created?